From d25d5ef7140ec64502296ce66cd953f840d25598 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:31:34 -0400
Subject: [alsa] usb-audio: Save mixer status only once at suspend

Message-id: <140972823573-285-patchset-jkysela@redhat.com>
Patchwork-id: 90981
O-Subject: [RHEL 7.1 PATCH 284/442] ALSA: usb-audio: Save mixer status only once at suspend
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The suspend callback of usb-audio driver may be called multiple times
per suspend when multiple USB interfaces are bound to a single sound
card instance.  In such a case, it's superfluous to save the mixer
values multiple times.  This patch fixes it by checking the counter.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 1c53e7253ed8769a00afa0f06777d731dbe1ba6f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/card.c b/sound/usb/card.c
index 893d5a1..e769d39 100644
--- a/sound/usb/card.c
+++ b/sound/usb/card.c
@@ -695,8 +695,9 @@ static int usb_audio_suspend(struct usb_interface *intf, pm_message_t message)
    chip->autosuspended = 1;
  }
 
- list_for_each_entry(mixer, &chip->mixer_list, list)
-  snd_usb_mixer_suspend(mixer);
+ if (chip->num_suspended_intf == 1)
+  list_for_each_entry(mixer, &chip->mixer_list, list)
+   snd_usb_mixer_suspend(mixer);
 
  return 0;
 }
-- 
1.7.1